client-tags/typing.md

```diff
@@ -27,14 +27,14 @@ For the purpose of this specification, a message is defined as a user-generated
 ## Format
 The `typing` tag is sent as a [`TAGMSG`][tags]. The client tag takes no parameters.
 
-The `typing` client tag SHOULD be sent by the client when the first character is typed in the text-input field. If new text is entered into the text-input field following sending of the text-input field content, the client MUST send a new `typing` tag after sending the preceding content.
+The `typing` client tag SHOULD be sent by the client when the first character is typed in the text-input field and after every 3 seconds of a non-empty text-input field to avoid a state change. If new text is entered into the text-input field following sending of the text-input field content, the client MUST send a new `typing` tag after sending the preceding content.
 
-After receiving a `typing` tag, the client SHOULD assume the state is unchanged until either a message end is received, or a period of 10 seconds has elapsed from when the last `typing` client tag was received and none of the previously defined message types were received in the appropriate target.
+After receiving a `typing` tag, the client SHOULD assume the state is unchanged until either a message end is received, or a period of 3 seconds has elapsed from when the last `typing` client tag was received and none of the previously defined message types were received in the appropriate target.
```

Comment:
The recipient timeout here should be longer than the throttling on the sender's end to allow for network delays. 6 seconds maybe?

```
After receiving a typing notification, clients SHOULD assume the sender is still typing until either a message is received, the sender leaves the channel or quits the server, or at least 6 seconds have elapsed since their last typing notification was received.
```

Again, this lets clients wait more than 6 seconds, but I think that's OK too.